features/storage/kvstore/kv_map/KVMap.h

```diff
@@ -0,0 +1,187 @@
+     *
+     * @return 0 on success, negative error code on failure
+     */
+    int deinit();
+
+    /**
+     * @brief full name lookup and then break it into KVStore instance and key
+     *
+     * @param full_name  string parameter contains the /partition name/key.
+     * @param kv_instance the main KVStore instance associated with the required partition name.
+     * @param key_index an index to the first character of the key.
+     * @return 0 on success, negative error code on failure
+     */
+    int lookup(const char *full_name, mbed::KVStore **kv_instance, size_t *key_index);
+
+    /**
+     * @brief full name lookup and then break it into KVStore instance and key
+     *
+     * @param name string parameter contains the /partition name/.
+     *
+     * @return pointer to the internal kvstore on success,
+     *         NULL on failure or if not exist
+     */
+    KVStore * get_internal_kv_instance(const char *name);
+    /**
+     * @brief full name lookup and then break it into KVStore instance and key
+     *
+     * @param name string parameter contains the /partition name/.
+     *
+     * @return pointer to the external kvstore on success,
+     *         NULL on failure or if not exist
+     */
+    KVStore * get_external_kv_instance(const char *name);
+    /**
+     * @brief full name lookup and then break it into KVStore instance and key
+     *
+     * @param name string parameter contains the /partition name/.
+     *
+     * @return pointer to the main kvstore on success,
+     *         NULL on failure or if not exist
+     */
+    KVStore * get_main_kv_instance(const char *name);
+    /**
+     * @brief full name lookup and then break it into KVStore instance and key
+     *
+     * @param name string parameter contains the /partition name/.
+     *
+     * @return pointer to the internal BlockDevice on success,
+     *         NULL on failure or if not exist
+     */
+    BlockDevice * get_internal_blockdevice_instance(const char *name);
+    /**
+     * @brief full name lookup and then break it into KVStore instance and key
+     *
+     * @param name string parameter contains the /partition name/.
+     *
+     * @return pointer to the external BlockDevice on success,
+     *         NULL on failure or if not exist
+     */
+    BlockDevice * get_external_blockdevice_instance(const char *name);
+    /**
+     * @brief full name lookup and then break it into KVStore instance and key or if not exist
+     *
+     * @param name string parameter contains the /partition name/.
+     *
+     * @return pointer to the external FileSystem on success,
+     *         NULL on failure or if not exist
+     */
+    FileSystem * get_external_filesystem_instance(const char *name);
+private:
+
+    /**
+     * @brief deinitialize  all component of a partition configuration struct.
+     *
+     * @param partition  partition configuration struct.
+     */
+    void deinit_partition(kv_map_entry_t * partition);
```

Comment:
`#if !defined(DOXYGEN_ONLY)` for private/protected members/methods in new files? so doxygen is not generated for our public docs (see https://github.com/ARMmbed/mbed-os/pull/8670)

Reply:
Done.